rust/lance/src/dataset.rs

```diff
@@ -1239,6 +1255,32 @@ impl Dataset {
         }
     }
 
+    pub async fn count_unindexed_rows(&self, index_name: &str) -> Result<Option<usize>> {
+        let indices = self.load_indices().await?;
+        let index = indices.iter().find(|idx| idx.name.eq(index_name));
+
+        if let Some(index) = index {
+            let unindexed_frags = unindexed_fragments(index, self).await?;
+            let unindexed_rows: usize = unindexed_frags.iter().filter_map(Fragment::num_rows).sum();
```

Comment:
There are a number of places where we added new features to the manifest gradually, by allowing the fields to not be present.  So, for example, `Fragment::num_rows` might return `None` to indicate that the manifest was generated by an older version of lance that didn't record the number of rows in the metadata.

Here, you are using `filter_map...sum`, which will basically treat `None` as `0`, which is potentially misleading.

I'm not sure if we should error in this case or return an option but the current approach could yield incorrect values which could be confusing.

Reply:
Ah I see now! I had made a bad assumption that `None` meant there was no deletion_file on the fragment, but hadn't considered there was fragments that didn't keep track of the number of rows. This should be fixed now